watertap/core/plugins/solvers.py

```diff
@@ -202,6 +206,84 @@ def _is_user_scaling(self):
         return True
 
 
+def create_debug_solver_wrapper(solver_name):
+
+    assert pyo.SolverFactory(solver_name).available()
+
+    debug_solver_name = f"debug-solver-wrapper-{solver_name}"
+
+    @pyo.SolverFactory.register(
+        debug_solver_name,
+        doc=f"Debug solver wrapper for {solver_name}",
+    )
+    class DebugSolverWrapper:
```

Comment:
I don't know if this is compatible with the way `@pyo.SolverFactory.register()` works, but would it be possible to move the class definition for `DebugSolverWrapper` in the outer (module) scope, and only do the registration (as a callable rather than as a decorator) inside `create_debug_solver_wrapper()`? e.g.

```py
class DebugSolverWrapper:
    ...


def create_debug_solver_wrapper(solver_name):
    DebugSolverWrapper._base_solver = solver_name
    debug_solver_name = ...
    pyo.SolverFactory.register(debug_solver_name, doc=...)(DebugSolverWrapper)
```

I don't think there'd be any significant change in functionality, but it'd be a bit more readable (and I think at least the class definition lines would be covered since they'd be in the module scope).

Reply:
I agree this is more readable, for two reasons:
1. removes a level of indentation
2. Makes it easier to reason about this generator function.

So I implemented it in 57e510f.